apps/api/src/service/application-service.ts

```diff
@@ -167,44 +172,92 @@ const applicationService = (db: PostgresDb) => ({
 			return failure(message, err);
 		}
 	},
+
 	listApplications: async ({
 		user_id,
-		state,
+		state = [],
 		sort = [],
 		page = 0,
 		pageSize = 20,
 	}: {
 		user_id?: string;
-		state?: ApplicationStateValues;
+		state?: ApplicationStateValues[];
 		sort?: Array<OrderBy<ApplicationsColumnName>>;
 		page?: number;
 		pageSize?: number;
 	}) => {
 		try {
-			const allApplications = await db
-				.select({
-					id: applications.id,
-					user_id: applications.user_id,
-					state: applications.state,
-					createdAt: applications.created_at,
-					updatedAt: applications.updated_at,
-				})
+			/**
+			 * Ensure that the page size or page somehow passed into here is not negative or not a number.
+			 * This should be handled at at the router layer, but just in-case.
+			 */
+			if (Number.isNaN(page) || Number.isNaN(pageSize)) {
+				throw Error('Page and/or page size must be a positive integer.');
+			} else if (page < 0 || pageSize < 0) {
+				throw Error('Page and/or page size must be non-negative values.');
+			}
+
+			const rawApplicationData = await db
+				.select()
 				.from(applications)
 				.where(
 					and(
 						user_id ? eq(applications.user_id, String(user_id)) : undefined,
-						state ? eq(applications.state, state) : undefined,
+						state.length ? inArray(applications.state, state) : undefined,
 					),
 				)
+				.leftJoin(applicationContents, eq(applications.contents, applicationContents.id))
 				.orderBy(...sortQuery(sort))
-				.offset(page * pageSize)
-				.limit(pageSize);
+				.offset(page * pageSize);
+
+			let allApplications = rawApplicationData.slice(page, pageSize).map((application) => {
```

Comment:
We can leverage destructuring here `.map(({applications, application_contents}) =>` and remove the repeat use of `application`